quickjs.c

```diff
@@ -38057,10 +38057,10 @@ static JSValue js_error_set_stackTraceLimit(JSContext *ctx, JSValue this_val, JS
 {
     if (JS_IsUndefined(this_val) || JS_IsNull(this_val))
         return JS_ThrowTypeErrorNotAnObject(ctx);
-    int limit;
-    if (JS_ToInt32(ctx, &limit, value) < 0)
+    double limit = 0;
+    if (JS_ToFloat64(ctx, &limit, value) < 0)
         return JS_EXCEPTION;
-    ctx->error_stack_trace_limit = limit;
+    ctx->error_stack_trace_limit = isfinite(limit) ? limit : INT32_MAX;
```

Comment:
I think you need to check for infinity and then for < 0 here. This test will fail now, and it shouldn't.

```
diff --git a/tests/bug858.js b/tests/bug858.js
index be66bd7..debc6b3 100644
--- a/tests/bug858.js
+++ b/tests/bug858.js
@@ -2,3 +2,7 @@ import {assert} from "./assert.js";
 
 Error.stackTraceLimit = Infinity;
 assert(new Error("error").stack.includes("bug858.js")); // stack should not be empty
+
+Error.stackTraceLimit = -Infinity;
+assert(!(new Error("error").stack.includes("bug858.js"))); // stack should be empty
+
```

v8 doesn't produce a stack trace in that case. Same goes for NaN. Let's add some tests for those as well please.

Reply:
Just tested now, the tests pass with NaN, Infinity, -Infinity.

Reply:
How did you test? My diff against your PR is what fails for me.